http/rest-client-reactive/src/main/java/io/quarkus/ts/http/restclient/reactive/fault/tolerance/HeaderResource.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.ts.http.restclient.reactive.fault.tolerance;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.MultivaluedMap;
+import jakarta.ws.rs.core.Response;
+
+/**
+ * Stores request headers from incoming server requests
+ * and exposes an endpoint that returns these headers
+ * It is used for testing purposes to verify if headers are correctly propagated.
+ */
+@Path("/fault/headers")
+@ApplicationScoped
+public class HeaderResource {
+
+    private final List<Map<String, String>> headerList = new ArrayList<>();
```

Comment:
strictly speaking your client is async so this needs to be thread-safe. e.g. use CopyOnWriteArrayList

Reply:
no, it doesn't work this way, it's ok, sorry